src/runtime_src/xocl/core/memory.cpp

```diff
@@ -333,7 +333,9 @@ get_memidx_nolock(const device* dev, memory::memidx_type subidx) const
   if (mset.none())
     throw std::runtime_error("No matching memory index");
 
-  for (size_t idx=0; idx<mset.size(); ++idx) {
+  // As connectivity section contains both group and bank index. Traverse from
+  // the higher order to give priority on group index over bank index
+  for (int idx=mset.size() - 1; idx >= 0; --idx) {
```

Comment:
Argh.  You should check if there are more places like this.  I see one in update_memidx_nolock(...) though I am not sure where and how it is used.

Reply:
Hi @stsoe ,
The other place which has a similar code i.e. default_allocation_message()
  ```
auto mask = device->get_boh_memidx(boh);
  xocl::device::memidx_type memidx = 0;
  for (size_t idx=0; idx<mask.size(); ++idx) {
    if (mask.test(idx)) {
      memidx = idx;
      break;
    }
  }
```
But I believe, this need not required any changes. For default memory bank we should give priority to the Banks.
Please share your comments.

Reply:
Hi @stsoe ,
I have updated the same for other places. Apart from that, there is another place which has a similar code i.e. default_allocation_message()

auto mask = device->get_boh_memidx(boh);
xocl::device::memidx_type memidx = 0;
for (size_t idx=0; idx<mask.size(); ++idx) {
  if (mask.test(idx)) {
    memidx = idx;
    break;
  }
}
But I believe, this need not required any changes. For default memory bank we should give priority to the Banks.
Please share your comments.